packages/contracts/deploy/40_finalize-managing-dao/10_install-multisig-on-managing-dao.ts

```diff
@@ -0,0 +1,123 @@
+    );
+
+  // Get `managingDAO` address.
+  const managingDAOAddress = await getContractAddress('DAO', hre);
+
+  // Get `DAO` contract.
+  const managingDaoContract = await ethers.getContractAt(
+    'DAO',
+    managingDAOAddress
+  );
+
+  // Get `PluginSetupProcessor` address.
+  const pspAddress = await getContractAddress('PluginSetupProcessor', hre);
+
+  // Get `PluginSetupProcessor` contract.
+  const pspContract = await ethers.getContractAt(
+    'PluginSetupProcessor',
+    pspAddress
+  );
+
+  // Installing multisig
+  const multisigRepoAddress = ehre.aragonPluginRepos.multisig;
+  const versionTag = [1, 1];
+  const pluginSetupRef = [versionTag, multisigRepoAddress];
+
+  // Prepare multisig plugin for managingDAO
+  const data = ethers.utils.defaultAbiCoder.encode(
+    buildMetadataJson.pluginSetupABI.prepareInstallation,
+    [approvers, [listedOnly, minApprovals]]
+  );
+  const prepareParams = [pluginSetupRef, data];
+  const prepareTx = await pspContract.prepareInstallation(
+    managingDAOAddress,
+    prepareParams
+  );
+  prepareTx.wait();
+
+  // extract info from prepare event
+  const event = await findEvent(prepareTx, 'InstallationPrepared');
+  const installationPreparedEvent = event.args;
+
+  // Grant `ROOT_PERMISSION` to `PluginSetupProcessor`.
+  await managePermission({
+    isGrant: true,
+    permissionManagerContract: managingDaoContract,
+    where: {name: 'DAO', address: managingDAOAddress},
+    who: {name: 'PluginSetupProcessor', address: pspAddress},
+    permission: 'ROOT_PERMISSION',
+  });
+
+  // Grant `APPLY_INSTALLATION_PERMISSION` to `Deployer`.
+  await managePermission({
+    isGrant: true,
+    permissionManagerContract: managingDaoContract,
+    where: {name: 'PluginSetupProcessor', address: pspAddress},
+    who: {name: 'Deployer', address: deployer},
+    permission: 'APPLY_INSTALLATION_PERMISSION',
+  });
+
+  // Apply multisig plugin to the managingDAO
+  const applyParams = [
+    pluginSetupRef,
+    installationPreparedEvent.plugin,
+    installationPreparedEvent.preparedSetupData.permissions,
+    hashHelpers(installationPreparedEvent.preparedSetupData.helpers),
+  ];
+  const applyTx = await pspContract.applyInstallation(
+    managingDAOAddress,
+    applyParams
+  );
+  applyTx.wait();
+
+  // Revoke `ROOT_PERMISSION` from `PluginSetupProcessor`.
+  await managePermission({
+    isGrant: false,
+    permissionManagerContract: managingDaoContract,
+    where: {name: 'DAO', address: managingDAOAddress},
+    who: {name: 'PluginSetupProcessor', address: pspAddress},
+    permission: 'ROOT_PERMISSION',
+  });
+
+  // Revoke `APPLY_INSTALLATION_PERMISSION` from `Deployer`.
+  await managePermission({
+    isGrant: false,
+    permissionManagerContract: managingDaoContract,
+    where: {name: 'PluginSetupProcessor', address: pspAddress},
+    who: {name: 'Deployer', address: deployer},
+    permission: 'APPLY_INSTALLATION_PERMISSION',
+  });
+};
```

Comment:
I'd add a check that the plugin (which is `multisig` in our case has `EXECUTE_PERMISSION` on the managing dao.

Btw, I don't like too much putting something in `hre` and getting it in another file from `hre`. doesn't there exist any other way ?